Symfony/CS/Tests/Fixer/Contrib/OrderedUseFixerTest.php

```diff
@@ -638,4 +653,723 @@ public function provide70Cases()
+use Foo\Bar;
+use Foo\Zar\Baz;
+use Acme\MyReusableTrait;
+use Symfony\Annotation\Template;
+use Foo\Bar\Foo as Fooo, Foo\Bir as FBB;
+use SomeClass;
+
+$a = new Bar();
+$a = new FooBaz();
+$a = new someclass();
+
+use Symfony\Doctrine\Entities\Entity;
+
+class AnnotatedClass
+{
+    use MyReusableTrait;
+
+    /**
+     * @Template(foobar=21)
+     * @param Entity $foo
+     */
+    public function doSomething($foo)
+    {
+        $bar = $foo->toArray();
+        /** @var ArrayInterface $bar */
+
+        return function () use ($bar, $baz) {};
+    }
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testByLengthFixWithDifferentCases()
+    {
+        OrderedUseFixer::configure(array(OrderedUseFixer::SORT_LENGTH));
+
+        $expected = <<<'EOF'
+The normal
+use of this fixer
+should not change this sentence nor those statements below
+use Zoo\Baz;
+use abc\Bar;
+
+<?php
+
+use abc\Bar;
+use Zoo\Baz;
+
+class Test
+{
+}
+EOF;
+
+        $input = <<<'EOF'
+The normal
+use of this fixer
+should not change this sentence nor those statements below
+use Zoo\Baz;
+use abc\Bar;
+
+<?php
+
+use Zoo\Baz;
+use abc\Bar;
+
+class Test
+{
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testByLengthOrderWithTrailingDigit()
+    {
+        OrderedUseFixer::configure(array(OrderedUseFixer::SORT_LENGTH));
+
+        $expected = <<<'EOF'
+<?php
+
+use abc\Bar;
+use abc2\Bar2;
+use xyz\abc\Bar6;
+use xyz\xyz\Bar4;
+use xyz\xyz\Bar5;
+use xyz\abc2\Bar7;
+
+class Test
+{
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+use abc2\Bar2;
+use abc\Bar;
+use xyz\abc2\Bar7;
+use xyz\abc\Bar6;
+use xyz\xyz\Bar4;
+use xyz\xyz\Bar5;
+
+class Test
+{
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testByLengthCodeWithImportsOnly()
+    {
+        OrderedUseFixer::configure(array(OrderedUseFixer::SORT_LENGTH));
+
+        $expected = <<<'EOF'
+<?php
+
+use Aaa;
+use Bbb;
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+use Bbb;
+use Aaa;
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testByLengthWithoutUses()
+    {
+        OrderedUseFixer::configure(array(OrderedUseFixer::SORT_LENGTH));
+
+        $expected = <<<'EOF'
+<?php
+
+$c = 1;
+EOF
+        ;
+
+        $this->makeTest($expected);
+    }
+
+    /**
+     * @dataProvider provide70CasesByLength
+     * @requires PHP 7.0
+     */
+    public function test70ByLength($expected, $input = null)
+    {
+        OrderedUseFixer::configure(array(OrderedUseFixer::SORT_LENGTH));
+        $this->makeTest($expected, $input);
+    }
+
+    public function provide70CasesByLength()
+    {
+        return array(
+            array(
+                '<?php
+use A\B;
+use function some\a\{fn_a, fn_b, fn_c};
+use some\a\{ClassA, ClassB, ClassC as C};
+use some\b\{ClassA, ClassB, ClassC as C};
+use const some\a\{ConstA, ConstB, ConstC};
+use const some\a\{ConstB, ConstA, ConstC};
```

Comment:
this example suggest you sort them by import type. is it? after deeper look it looks like coincidence. create more complex case.